captum/_utils/gradient.py

```diff
@@ -32,7 +32,14 @@ def apply_gradient_requirements(inputs: Tuple[Tensor, ...]) -> List[bool]:
     for index, input in enumerate(inputs):
         assert isinstance(input, torch.Tensor), "Given input is not a torch.Tensor"
         grad_required.append(input.requires_grad)
-        if not input.requires_grad:
+        inputs_dtype = input.dtype
+        if inputs_dtype == torch.long or inputs_dtype == torch.int:
```

Comment:
Yeah, it didn't cover all cases - just the common ones.
It looks like in python API we can check on the floating point and complex types:
https://github.com/pytorch/pytorch/blob/master/torch/tensor.py#L547
This can potentially suffice.